Sources/Swift/Metrics/BucketsMetricsAggregator.swift

```diff
@@ -20,6 +20,7 @@ class BucketMetricsAggregator: MetricsAggregator {
     private let currentDate: SentryCurrentDateProvider
     private let dispatchQueue: SentryDispatchQueueWrapper
     private let random: SentryRandomProtocol
+    private let beforeEmitMetric: BeforeEmitMetricCallback
```

Comment:
l: IMO we should have nullable callbacks, so we dont need to create blocks that dont do anything as it happens twice in this PR.